src/error.rs

```diff
@@ -1,11 +1,118 @@
 //! Various error types used thorough the crate.
-
-pub use self::{
-    compile::Error as CompileError, jit_engine::Error as JitRunnerError,
-    libfuncs::Error as CoreLibfuncBuilderError, types::Error as CoreTypeBuilderError,
+use cairo_lang_sierra::{
+    edit_state::EditStateError, ids::ConcreteTypeId, program_registry::ProgramRegistryError,
 };
 
-pub mod compile;
-pub mod jit_engine;
-pub mod libfuncs;
-pub mod types;
+use std::{alloc::LayoutError, fmt, num::TryFromIntError, ops::Deref};
+use thiserror::Error;
+
+use crate::metadata::gas::GasMetadataError;
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug, Error)]
+pub struct Error {
+    pub source: Box<ErrorImpl>,
```

Comment:
i think Error itself can be the ErrorImpl, and if possible remove the box

Reply:
I've just checked it and it seems `std::backtrace::Backtrace` is stable since Rust 1.65.0.
If you're going to add the error location backtraces please add them in `Error`, otherwise there's no point to having `Error` wrap `ErrorImpl` as @edg-l said.

Reply:
I tried to do this but the same errors pop up and Dtolnay says that its still not ready to be used with his package. So I've gone and removed the errorimp instead. I am on rust 1.76.